metpy/calc/thermo.py

```diff
@@ -1711,3 +1711,155 @@ def thickness_hydrostatic_from_relative_humidity(pressure, temperature, relative
 
     return thickness_hydrostatic(pressure, temperature, mixing=mixing, bottom=bottom,
                                  depth=depth)
+
+
+@exporter.export
+@check_units('[length]', '[temperature]')
+def brunt_vaisala_frequency_squared(heights, potential_temperature, axis=0):
+    r"""Calculate the square of the Brunt-Vaisala frequency.
+
+    Brunt-Vaisala frequency squared (a measure of atmospheric stability) is given by the
+    formula:
+
+    .. math:: N^2 = \frac{g}{\theta} \frac{d\theta}{dz}
+
+    This formula is based off of Equations 3.75 and 3.77 in [Hobbs2006]_.
+
+    Parameters
+    ----------
+    heights : array-like
+        One-dimensional profile of atmospheric height
+    potential_temperature : array-like
+        Atmospheric potential temperature
+    axis : int, optional
+        The axis corresponding to vertical in the potential temperature array, defaults to 0.
+
+    Returns
+    -------
+    array-like
+        The square of the Brunt-Vaisala frequency.
+
+    See Also
+    --------
+    brunt_vaisala_frequency, brunt_vaisala_frequency_layer, brunt_vaisala_period,
+    potential_temperature
+
+    """
+    # Ensure validity of temperature units
+    potential_temperature.ito('K')
```

Comment:
Let's not use `ito` here since that modifies in place the data we were given--that could be very surprising to users. Try instead:
```python
potential_temperature = potential_temperature.to('K')
```